src/ssl.c

```diff
@@ -1474,16 +1474,34 @@ WOLFSSL* wolfSSL_new(WOLFSSL_CTX* ctx)
 
     WOLFSSL_ENTER("wolfSSL_new");
 
-    if (ctx == NULL)
+    if (ctx == NULL) {
+        WOLFSSL_MSG("wolfSSL_new ctx is null");
         return ssl;
+    }
 
     ssl = (WOLFSSL*) XMALLOC(sizeof(WOLFSSL), ctx->heap, DYNAMIC_TYPE_SSL);
     if (ssl) {
-        if ( (ret = InitSSL(ssl, ctx, 0)) < 0) {
+        ret = InitSSL(ssl, ctx, 0);
+        if (ret < 0) {
+            WOLFSSL_MSG_EX("wolfSSL_new failed during InitSSL. ret = %d", ret);
             FreeSSL(ssl, ctx->heap);
             ssl = 0;
         }
-    }
+        else {
```

Comment:
make this 
```
else if (ret == 0) {
...
}
else {
...
}
```

Reply:
Agreed. Your suggestion results in improved readability. Done